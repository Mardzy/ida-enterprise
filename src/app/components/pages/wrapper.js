import React from "react";
import { Switch, Route } from "react-router";

import ContentWithSidebar from "../../../ui/layout/content-with-sidebar";

import Chart from "../../../modules/chart";
import Dashboard from "../../../modules/chart-group";
import MainMenu from "../../../modules/navigation";

/**
 * Render a chart.
 * @param {*} chartId
 * @return {*}
 */
const renderChart = chartId => <Chart key={chartId} id={chartId}/>;

/**
 * Render nested routes.
 * @return {*}
 */
const renderContent = () => (
  <Switch>
    <Route render={() => <Dashboard pageId="dashboard" renderChart={renderChart}/>}/>
  </Switch>
);

/**
 * Render main menu on sidebar.
 * @return {*}
 */
const renderSidebar = () => <MainMenu/>;

/**
 * Page wrapper with navigation.
 * @type {ReactComponent}
 */
const WrapperWithNavigation = () => <ContentWithSidebar padded sidebar={renderSidebar} content={renderContent}/>;

export default WrapperWithNavigation;
