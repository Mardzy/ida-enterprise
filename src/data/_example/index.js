import reducer from "./reducers";

export { default as duckEpics } from "./epics";
export { default as duckTypes } from "./types";

export default reducer;