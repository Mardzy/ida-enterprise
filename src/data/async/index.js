import { async as reducer } from "./reducers";

export default reducer;