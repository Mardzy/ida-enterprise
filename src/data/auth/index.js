import authReducer from "./reducers";

export { default as authEpics } from "./epics";
export { default as authTypes } from "./types";
export { default as authActions } from "./actions";

export default authReducer;