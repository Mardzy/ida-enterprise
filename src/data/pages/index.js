import pagesReducer from "./reducers";

export { default as pageEpics } from "./epics";
export { default as pageTypes } from "./types";

export default pagesReducer;