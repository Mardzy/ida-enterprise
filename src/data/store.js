import { fromJS } from "immutable";
import { createEpicMiddleware } from "redux-observable";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import persistState from "redux-localstorage";
import { routerMiddleware } from "connected-react-router/immutable";

import { INITIAL_STATE } from "./initial-state";
import { rootReducer, history } from "./reducer";
import { rootEpic } from "./epic";
// import { gaMiddleware } from "data/config/analytics";
import { serialize, deserialize, merge, slice } from "./config/persistent-store";

/**
 * Construct an epic middleware.
 * @type {EpicMiddleware<Action, any, any, Action>}
 */
const epicMiddleware = createEpicMiddleware();

/**
 * Middleware to apply in the store.
 * @type {*[]}
 */
const middleware = [
  // gaMiddleware,
  epicMiddleware,
  routerMiddleware(history)
];

/**
 * Configuration for keeping all the redux state in localstorage.
 * @type {{key: string, slicer: slice, serialize: serialize, deserialize: deserialize, merge: merge}}
 */
const persistStateConfig = {
  key: "partner-dashboard",
  slicer: slice,
  serialize, deserialize, merge
};

/**
 * Create a Redux store.
 * @param {Object} initialState
 * @return {Store<any>}
 */
const configureStore = (initialState = {}) => createStore(
  rootReducer,
  INITIAL_STATE.merge(fromJS(initialState)),
  composeWithDevTools({})(
    applyMiddleware(...middleware),
    persistState(["auth", "farms"], persistStateConfig)
  )
);

/**
 * Create a store with the reducers.
 */
const store = configureStore();

epicMiddleware.run(rootEpic);

export default store;
export { history, epicMiddleware, configureStore };