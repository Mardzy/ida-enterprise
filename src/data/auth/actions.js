import types from "./types";
import { createAsyncActions } from "../async/utils";

/**
 * Login actions created from types.LOGIN
 * @type {{start: Action, fulfilled: Action, rejected: Action}}
 */
const login = createAsyncActions(
  types.LOGIN,
  (username, password) => ({
    username,
    password
  }),
  response => response,
  response => response
);


/**
 * Logout action
 * @type {Action}
 * @memberOf module:Actions
 * @return {{type: string}}
 */
const logout = () => ({
  type: types.LOGOUT.PENDING
});

/**
 * Actions created from types.REQUEST_USER
 * @type {{start: Action, fulfilled: Action, rejected: Action}}
 */
const requestUser = createAsyncActions(
  types.REQUEST_USER,
  username => ({
    email: username
  }),
  response => response,
  response => response
);

export default {
  login,
  logout,
  requestUser
};

