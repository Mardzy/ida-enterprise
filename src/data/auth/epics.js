import _ from "lodash";
import { of } from "rxjs";
import { ajax } from "rxjs/ajax";
import { mergeMap, mapTo, map, catchError } from "rxjs/operators";
import { ofType } from "redux-observable";

import { IDA_ENDPOINTS } from "../config/services";
import types from "./types";
import actions from "./actions";
import { Request } from "../api/ajax";

/**
 * Handle errors during login.
 * @param error
 * @return {Observable}
 */
const handleLoginError = error => of(actions.login.rejected(error));

/**
 * Register the token for further data requests, then pass on to the successful login handler.
 * @param request
 * @return {{type: string, payload: {response: Object}}}
 */
const handleLoginSuccess = request => {
  Request.registerToken(request.response.access_token);

  return actions.login.fulfilled(request.response);
};

/**
 * Login request config
 * @type {{method: string, url: string, crossDomain: boolean, headers: {}}}
 */
const loginRequestConfig = {
  method: "POST",
  url: `${IDA_ENDPOINTS.auth}/token`,
  crossDomain: true,
  headers: {
    "content-type": "application/x-www-form-urlencoded"
  }
};

/**
 * Log the user in, then pass the appropriate answer back to Redux.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 */
const login = action$ => action$.pipe(
  ofType(types.LOGIN.PENDING),
  mergeMap(action =>
    ajax(
      _.assign({}, loginRequestConfig, {
        body: _.assign({ grant_type: "password" }, action.payload)
      })
    ).pipe(
      map(handleLoginSuccess),
      catchError(handleLoginError)
    )
  )
);

/**
 * Logs the user out.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 * @return {Observable<{type: string}>}
 */
const logout = action$ => action$.pipe(
  ofType(types.LOGOUT.PENDING),
  mapTo({
    type: types.LOGOUT.FULFILLED
  })
);

/**
 * Handle errors during requesting user information.
 * @param error
 * @return {Observable}
 */
const handleUserRequestError = error => of(actions.requestUser.rejected(error));

/**
 * Store user information after successful request.
 * @param request
 * @return {{type: string, payload: {response: Object}}}
 */
const handleUserRequestSuccess = request => actions.requestUser.fulfilled(request.response);

/**
 * Request a user.
 * @type {Epic}
 * @memberOf module:Epics
 * @param action$
 * @return {Observable<AjaxResponse>}
 */
const requestUser = action$ => action$.pipe(
  ofType(types.REQUEST_USER.PENDING),
  mergeMap(action =>
    Request.GET(
      `${IDA_ENDPOINTS.auth}/Users/Find/EmailSimple/${encodeURIComponent(action.payload.email)}`
    ).pipe(
      map(handleUserRequestSuccess),
      catchError(handleUserRequestError)
    )
  )
);

export default {
  login,
  logout,
  requestUser
};