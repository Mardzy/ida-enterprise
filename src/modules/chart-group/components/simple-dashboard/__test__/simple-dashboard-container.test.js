import { Provider } from "react-redux";

import store from "../../../../../data/store";
import { authTypes } from "../../../../../data/auth";
import { pageTypes } from "../../../../../data/pages";

import SimpleDashboardContainer from "../simple-dashboard-container";

describe("The simple dashboard container", () => {

  it("renders a spinner while loading", () => {
    const wrapper = mount(
      <Provider store={store}>
        <SimpleDashboardContainer/>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the section contents from the store", () => {
    /**
     * @type {string}
     */
    const pageId = "dashboard";

    /**
     * @param chartId
     */
    const renderChart = chartId => expect(typeof chartId).toBe("number");

    /**
     * @type {[]}
     */
    const charts = [{ id: 1, config: { type: "geo"} }, { id: 2, config: { type: "stats"} }];

    store.dispatch({
      type: authTypes.REQUEST_USER.FULFILLED,
      payload: {
        givenName: "Sandor",
        surName: "Kocsis"
      }
    });

    store.dispatch({
      type: pageTypes.FETCH_CHART_LIST.FULFILLED,
      payload: { pageId, layout: charts }
    });

    const wrapper = mount(
      <Provider store={store}>
        <SimpleDashboardContainer renderChart={renderChart} pageId={pageId}/>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  })

});