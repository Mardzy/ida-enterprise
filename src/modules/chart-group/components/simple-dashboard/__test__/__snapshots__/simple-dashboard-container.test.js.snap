// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The simple dashboard container renders a spinner while loading 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(withProps(lifecycle(branch(Sink))))>
    <withProps(lifecycle(branch(Sink)))
      fetchData={[Function]}
      isLoading={true}
    >
      <lifecycle(branch(Sink))
        content={[Function]}
        fetchData={[Function]}
        isLoading={true}
      >
        <branch(Sink)
          content={[Function]}
          fetchData={[Function]}
          isLoading={true}
        >
          <renderComponent(SpinnerWithText)
            content={[Function]}
            fetchData={[Function]}
            isLoading={true}
          >
            <SpinnerWithText
              content={[Function]}
              fetchData={[Function]}
              isLoading={true}
            >
              <SpinnerWithOverlay>
                <Dimmer
                  active={true}
                  inverted={true}
                >
                  <DimmerInner
                    active={true}
                    inverted={true}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <div
                            class="ui active transition visible inverted dimmer"
                            style="display: flex;"
                          >
                            <div
                              class="content"
                            >
                              <div
                                class="ui inverted text loader"
                              >
                                Loading charts
                              </div>
                            </div>
                          </div>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <div
                              class="ui active transition visible inverted dimmer"
                              style="display: flex;"
                            >
                              <div
                                class="content"
                              >
                                <div
                                  class="ui inverted text loader"
                                >
                                  Loading charts
                                </div>
                              </div>
                            </div>,
                          }
                        }
                      >
                        <div
                          className="ui active transition visible inverted dimmer"
                          onClick={[Function]}
                        >
                          <div
                            className="content"
                          >
                            <Loader
                              inverted={true}
                            >
                              <div
                                className="ui inverted text loader"
                              >
                                Loading charts
                              </div>
                            </Loader>
                          </div>
                        </div>
                      </RefFindNode>
                    </Ref>
                  </DimmerInner>
                </Dimmer>
              </SpinnerWithOverlay>
            </SpinnerWithText>
          </renderComponent(SpinnerWithText)>
        </branch(Sink)>
      </lifecycle(branch(Sink))>
    </withProps(lifecycle(branch(Sink)))>
  </Connect(withProps(lifecycle(branch(Sink))))>
</Provider>
`;

exports[`The simple dashboard container renders the section contents from the store 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(withProps(lifecycle(branch(Sink))))
    pageId="dashboard"
    renderChart={[Function]}
  >
    <withProps(lifecycle(branch(Sink)))
      charts={
        Immutable.List [
          1,
          2,
        ]
      }
      fetchData={[Function]}
      isLoading={false}
      pageId="dashboard"
      renderChart={[Function]}
    >
      <lifecycle(branch(Sink))
        charts={
          Immutable.List [
            1,
            2,
          ]
        }
        content={[Function]}
        fetchData={[Function]}
        isLoading={false}
        pageId="dashboard"
        renderChart={[Function]}
      >
        <branch(Sink)
          charts={
            Immutable.List [
              1,
              2,
            ]
          }
          content={[Function]}
          fetchData={[Function]}
          isLoading={false}
          pageId="dashboard"
          renderChart={[Function]}
        >
          <renderComponent(SingleColumnWithFilteredProps)
            charts={
              Immutable.List [
                1,
                2,
              ]
            }
            content={[Function]}
            fetchData={[Function]}
            isLoading={false}
            pageId="dashboard"
            renderChart={[Function]}
          >
            <SingleColumnWithFilteredProps
              charts={
                Immutable.List [
                  1,
                  2,
                ]
              }
              content={[Function]}
              fetchData={[Function]}
              isLoading={false}
              pageId="dashboard"
              renderChart={[Function]}
            >
              <SingleColumn
                content={[Function]}
              >
                <Grid
                  columns={1}
                  stackable={true}
                >
                  <div
                    className="ui stackable one column grid"
                  >
                    <GridColumn>
                      <div
                        className="column"
                      >
                        <Component />
                      </div>
                    </GridColumn>
                  </div>
                </Grid>
              </SingleColumn>
            </SingleColumnWithFilteredProps>
          </renderComponent(SingleColumnWithFilteredProps)>
        </branch(Sink)>
      </lifecycle(branch(Sink))>
    </withProps(lifecycle(branch(Sink)))>
  </Connect(withProps(lifecycle(branch(Sink))))>
</Provider>
`;
